<launch>

    <!-- Modify here -->
    <arg name="gui" default="false" /> <!-- Use toolBar to set each joint angle. -->
    <arg name="Launch_Rviz" default="false"/> <!-- Use rviz -->


    <!-- Don't modify any command below !!! -->
    <arg name="model" default="$(find robot_arm_control)/urdf/Robot_Arm.urdf"/>
    <arg name="rvizconfig" default="$(find robot_arm_control)/rviz/urdf.rviz" />

    <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />

    <node if="$(arg gui)" name="joint_state_publisher" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" />
    
    <node unless="$(arg gui)" name="Rviz_Arm_Position_node" pkg="robot_arm_control" type="Rviz_Arm_Position_node" output="screen">
        <param name="PubRobotArmState" type="bool" value="false"/>
        <param name="Frequency"        type="int"  value="50"/>
    </node>

    <node name="GetObject_server" pkg="robot_arm_control" type="GetObject_server" output="screen">
        <param name="Param_RobotArm_GrabAngle_Open"  type="int" value="100"/>
        <param name="Param_RobotArm_GrabAngle_Close" type="int" value="120"/>

        <param name="Param_RobotArm_PutAngle_0"      type="int" value="75"/>
        <param name="Param_RobotArm_PutAngle_1"      type="int" value="200"/>
        <param name="Param_RobotArm_PutAngle_2"      type="int" value="250"/>
        
    </node>

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

    <node if="$(arg Launch_Rviz)" name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

</launch>
